### PR TITLE
Fix persistence cluster sharding serialization failure

### PR DESCRIPTION
**Pull request by Arkatufus**
Related to #5116

### CODE REVIEW

File: src/contrib/cluster/Akka.Cluster.Sharding/Shard.cs [hunk 1]
```diff
@@ -295,7 +295,7 @@ public CurrentShardState(string shardId, IImmutableSet<string> entityIds)
         /// TBD
         /// </summary>
         [Serializable]
-        public sealed class GetShardStats : IShardQuery
+        public sealed class GetShardStats : IShardQuery, IClusterShardingSerializable
```
Comment (Aaronontheweb) [inline src/contrib/cluster/Akka.Cluster.Sharding/Shard.cs:298]: Neither of these messages are persisted, so they couldn't cause the persistence failures included in https://github.com/akkadotnet/akka.net/issues/5116

**Review by Aaronontheweb — CHANGES_REQUESTED** [review]
Does not fix the issue

File: src/contrib/cluster/Akka.Cluster.Sharding/Shard.cs (hunk 1 above)
Comment (Arkatufus) [reply src/contrib/cluster/Akka.Cluster.Sharding/Shard.cs:298]: They should be included anyway:
https://github.com/akkadotnet/akka.net/blob/dev/src/contrib/cluster/Akka.Cluster.Sharding/Serialization/ClusterShardingMessageSerializer.cs#L155-L156

Comment (Aaronontheweb) [reply src/contrib/cluster/Akka.Cluster.Sharding/Shard.cs:298]: Yep, that's a good fix but it doesn't close the issue.

Comment (Aaronontheweb) [reply src/contrib/cluster/Akka.Cluster.Sharding/Shard.cs:298]: Do you want me to merge this issue separately?

Comment (Aaronontheweb) [reply src/contrib/cluster/Akka.Cluster.Sharding/Shard.cs:298]: I thought the original header for this issue said "close #5116"

Comment (Arkatufus) [reply src/contrib/cluster/Akka.Cluster.Sharding/Shard.cs:298]: no, its only related, found this problem while checking the serializer code

**Review by Aaronontheweb — APPROVED** [review]

**Pull request merged**